GenderPayGap.Core/Helpers/ReportingYearsHelper.cs

```diff
@@ -1,77 +1,104 @@
-        public static DateTime GetDeadlineForAccountingDate(DateTime accountingDate)
-        {
-            int reportingYear = accountingDate.Year;
-            DateTime deadline = accountingDate.AddYears(1).AddDays(-1);
-            if (reportingYear == 2020)
-            {
-                // Reporting deadline for 2020 was changed to 2021/10/05 for both public and private
-                deadline = new DateTime(2021,10,5);
-            }
-
-            return deadline;
-        }
-
-        public static int GetTheMostRecentCompletedReportingYear()
-        {
-            int mostRecentReportingYear = Global.FirstReportingYear;
-            foreach (int year in from year in GetReportingYears()
-                let accountingDate = SectorTypes.Private.GetAccountingStartDate(year)
-                where DeadlineForAccountingDateHasPassed(accountingDate)
-                select year)
-            {
-                mostRecentReportingYear = year;
-                break;
-            }
-
-            return mostRecentReportingYear;
-        }
-
-        public static bool DeadlineForAccountingDateHasPassed(DateTime accountingDate)
-        {
-            return GetDeadlineForAccountingDate(accountingDate).AddDays(1) < VirtualDateTime.Now;
-        }
-
-        public static bool IsReportingYearWithFurloughScheme(DateTime accountingDate)
-        {
-            return IsReportingYearWithFurloughScheme(accountingDate.Year);
-        }
-        
-        public static bool IsReportingYearWithFurloughScheme(int reportingYear)
-        {
-            return Global.ReportingStartYearsWithFurloughScheme.Contains(reportingYear);
-        }
-
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using GenderPayGap.Core.Classes;
+using GenderPayGap.Extensions;
+
+namespace GenderPayGap.Core.Helpers
+{
+    public static class ReportingYearsHelper
+    {
+        public static List<int> GetReportingYears()
+        {
+            int firstReportingYear = Global.FirstReportingYear;
+            int currentReportingYear = SectorTypes.Public.GetAccountingStartDate().Year;
+            int numberOfYears = currentReportingYear - firstReportingYear + 1;
+
+            List<int> reportingYears = Enumerable.Range(firstReportingYear, numberOfYears).Reverse().ToList();
+            return reportingYears;
+        }
+
+        public static List<(int,string)> GetReportingYearWithCSVFileSize()
+        {
+            List<int> reportingYears = GetReportingYears().OrderByDescending(y => y).ToList<int>();
+            List<(int, int)> yearsWithFileSizes = new List<(int, int)>()
+            {
+                (reportingYears[0], 55),
+                (reportingYears[1], 4032),
+                (reportingYears[2], 4129),
+                (reportingYears[3], 2798),
+                (reportingYears[4], 4266),
+                (reportingYears[5], 4211),
+            };

```

Comment:
Errrrm, it looks awfully like the file sizes here are hard-coded! 😱
Over time, as more organisations report their GPG, these file sizes will change.

Also, this method will only ever return 6 items.
Currently, there are 6 reporting years, but once we move into the next reporting year, there will be 7.
And it's wrong to not show all available files.

If we're going to fix a minor usability issue like this (adding file sizes), it's extra important not to break any existing functionality.

CC: @davidm-m